plugins/finder/jevents/jevents.php

```diff
@@ -384,19 +346,167 @@ protected function getItem($id)
 	}
 
+	/**
+	 * Overload method to update index data on published state changes
+	 *
+	 * @param   array    $pks    A list of primary key ids of the content that has changed state.
+	 * @param   integer  $value  The value of the state that the content has been changed to.
+	 *
+	 * @return  void
+	 *
+	 * @since   2.5
+	 */
+	protected function itemStateChange($pks, $value)
+	{
+		/*
+		 * The item's published state is tied to the category
+		 * published state so we need to look up all published states
+		 * before we change anything.
+		 */
+		foreach ($pks as $pk)
+		{
+			$query = clone $this->getStateQuery();
+			$query->where('a.ev_id = ' . (int) $pk);
+
+			// Get the published states.
+			$this->db->setQuery($query);
+			$item = $this->db->loadObject();
+
+			// Translate the state.
+			$temp = $this->translateState($value, $item->cat_state);
+
+			// Update the item.
+			$this->change($pk, 'state', $temp);
+
+			// Reindex the item
+			$this->reindex($pk);
+		}
+	}
+
+	/**
+	 * Overload method to remove an item from the index.
+	 *
+	 * @param   string  $id  The ID of the item to remove.
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @since   2.5
+	 * @throws  Exception on database error.
+	 */
+	protected function remove($id)
+	{
+		// set the item URL overriding $this->getURL() since JEvents is not using Joomla standard view & layout
+		$url    = 'index.php?option=' . $this->extension . '&task=icalevent.detail&evid=' . $id;
+
+		// Get the link ids for the content items.
+		$query = $this->db->getQuery(true)
+			->select($this->db->quoteName('link_id'))
+			->from($this->db->quoteName('#__finder_links'))
+			->where($this->db->quoteName('url') . ' LIKE ' . $this->db->quote('%' . $url . '%'));
+		$this->db->setQuery($query);
+		$items = $this->db->loadColumn();
+
+		// Check the items.
+		if (empty($items)) {
+			return true;
+		}
+
+		// Remove the items.
+		foreach ($items as $item) {
+			$this->indexer->remove($item);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Custom method to get the event repeat data.
+	 * This method will only get the first row Start Repeat Date and the Last Row End Repeat Date.
+	 * The dates will be use to populate the publish_start_date, publish_end_date, start_date, end_date.
+	 *
+	 * @param   void
+	 *
+	 * @since   1.0
+	 * @return  Query string.
+	 */
+	protected function getRepeats()
+	{
+		$db     = $this->db;
+		$query  = $db->getQuery(true);
+
+		$query->select('MIN(r.startrepeat) AS start_repeat, MAX(r.endrepeat) AS end_repeat')
+			->from($db->quoteName('#__jevents_repetition', 'r'))
+			->group($db->quoteName('r.eventid'));
+
+		return $query;
+	}
+
+	/**
+	 * Custom method to get a custom field (keyword) for searching an event
+	 *
+	 * @param   $keyword    string  The custom field name.
+	 * @param   $id         int     The event id.
+	 *
+	 * @since   1.0
+	 * @return  Custom field value
+	 */
+	protected function getKeyword($customfield, $id)
```

Comment:
This method is called in the index() method, again customized for our needs and can be deleted. - Jackson